test/Dialect/Torch/simplify-dtype-calculations.mlir

```diff
@@ -0,0 +1,282 @@
+// RUN: torch-mlir-opt -torch-simplify-dtype-calculations -split-input-file %s | FileCheck %s
+
+// CHECK-LABEL:   func.func @basic(
+// CHECK-SAME:                     %[[ARG:.*]]: !torch.vtensor<*,f32>) -> !torch.vtensor {
+// CHECK:           %[[DTYPE_INT:.*]] = torch.constant.int 6
+// CHECK:           %[[RESULT:.*]] = torch.dtype.calculate {
+// CHECK:             %[[TANH:.*]] = torch.aten.tanh %[[ARG]] : !torch.vtensor<*,f32> -> !torch.vtensor<*,f32>
+// CHECK:             torch.dtype.calculate.yield %[[TANH]] : !torch.vtensor<*,f32>
+// CHECK:           } dtypes {
+// CHECK:             torch.dtype.calculate.yield.dtypes %[[DTYPE_INT]] : !torch.int
+// CHECK:           } : !torch.vtensor<*,f32>
+// CHECK:           %[[CAST:.*]] = torch.tensor_static_info_cast %[[RESULT]] : !torch.vtensor<*,f32> to !torch.vtensor
+// CHECK:           return %[[CAST]] : !torch.vtensor
+func.func @basic(%arg0: !torch.vtensor<*,f32>) -> !torch.vtensor {
+  %0 = torch.dtype.calculate {
+    %1 = torch.aten.tanh %arg0 : !torch.vtensor<*,f32> -> !torch.vtensor
+    torch.dtype.calculate.yield %1 : !torch.vtensor
+  } dtypes {
+    %2 = torch.prim.dtype %arg0 : !torch.vtensor<*,f32> -> !torch.int
+    torch.dtype.calculate.yield.dtypes %2 : !torch.int
+  } : !torch.vtensor
+  return %0 : !torch.vtensor
+}
+
+// -----
+
+// CHECK-LABEL:   func.func @promote_dtypes$tensor_tensor_same_category_different_width(
+// CHECK:             {{.*}} = torch.aten.add.Tensor {{.*}} -> !torch.vtensor<[1],f64>
+func.func @promote_dtypes$tensor_tensor_same_category_different_width(%arg0: !torch.vtensor<[1],f32>, %arg1: !torch.vtensor<[1],f64>, %arg2: !torch.float) {
+  %int7 = torch.constant.int 7
+  %int6 = torch.constant.int 6
+  %int1 = torch.constant.int 1
+  %0 = torch.dtype.calculate {
+    %1 = torch.aten.add.Tensor %arg0, %arg1, %arg2 : !torch.vtensor<[1],f32>, !torch.vtensor<[1],f64>, !torch.float -> !torch.vtensor<[1],unk>
+    torch.dtype.calculate.yield %1 : !torch.vtensor<[1],unk>
+  } dtypes {
+    %1 = torch.prim.ListConstruct %int1, %int1 : (!torch.int, !torch.int) -> !torch.list<optional<int>>
```

Comment:
can you explain what %int1/etc. represent? Even better, get the dtype directly off of a tensor with prim::dtype.